ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -354,9 +354,14 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
         return false;
       }
       const rp = this.props.workspace.researchPurpose;
-      return rp.ancestry || rp.controlSet ||
-          rp.diseaseFocusedResearch || rp.ethics || rp.drugDevelopment ||
-          rp.methodsDevelopment || rp.populationHealth || rp.socialBehavioral;
+      return this.researchPurposeCategoriesSelected(rp);
+    }
+
+    researchPurposeCategoriesSelected(researchPurpose) {
```

Comment:
can we add type signatures? I think it would help clear up a question I had below